Loop/Managers/Alerts/StoredAlert+CoreDataClass.swift

```diff
@@ -9,9 +9,23 @@
 
 import Foundation
 import CoreData
-
+import LoopKit
 
 public class StoredAlert: NSManagedObject {
+    
+    var interruptionLevel: Alert.InterruptionLevel {
+        get {
+            willAccessValue(forKey: "interruptionLevel")
+            defer { didAccessValue(forKey: "interruptionLevel") }
+            return try! Alert.InterruptionLevel(storedValue: primitiveValue(forKey: "interruptionLevel") as! Int16)
```

Comment:
The practice I've seen in other ManagedObjects is to have the primitive specified as `primitiveWhatever` and access that rather than using `primitiveValue(forKey:)` and `setPrimitiveValue(forKey:)`. See `CachedInsulinDeliveryObject` for an example.

Reply:
I did that exact thing at first...but I kept getting mysterious `EXC_BAD_ACCESS` crashes immediately in the setter.  I did some research but could not find a root cause; I got as far as finding out it had something to do with the autogenerated code from `@NSManaged`. So I removed it and implemented it "by hand" using the `primitiveValue` and `setPrimitiveValue` primitives.  The docs really don't encourage this, but I couldn't find another way.  Any ideas?

Reply:
What type did you use in the `StoredAlert+CoreDataProperties.swift` file? I believe it has to be `NSNumber`.

Reply:
Ah, ok.  That's the problem, then.  I used Int16...and TBH now this is ringing a bell.  It all seems rather silly to use @NSManaged, but then have to write custom code for the primitive type, which can't really be primitive but an NSNumber.  🙄 

I'll switch it to NSNumber.